### PR TITLE
Feat: Adds `SearchInput`, `SearchInputField` and `SearchDropdown` components

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Adds `SearchInput`, `SearchInputField` and `SearchDropdown` components to `faststore/ui` & `faststore/components`
- Refactor `SearchInput` on `faststore/core`
- Removes `SearchSuggestions` from `faststore/core`
- Adds `SearchInput`, `SearchInputField` and `SearchDropdown` components docs pages
- Adds `Search` feature page

Todo:

- [x] Adds Search Top props to doc page
- [x]  Adds Search History props to doc page
- [x] Adds Search Products props to doc page
- [x] Adds Search Auto Complete props to doc page

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/24

### CODE REVIEW

File: packages/components/src/molecules/SearchDropdown/SearchDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  },
+  ref
+) {
+  return (
+    <div ref={ref} data-fs-search-dropdown data-testid={testId} {...otherProps}>
+      {(() => {
+        if (term.length === 0 && !isLoading) {
+          return (
+            <>
+              {searchHistoryComponent}
```
Comment (ArthurTriis1) [inline packages/components/src/molecules/SearchDropdown/SearchDropdown.tsx:68]:
```suggestion
              <SearchHistoryComponent />
```

File: packages/ui/src/components/molecules/SearchDropdown/styles.scss [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      border: var(--fs-search-dropdown-border-width) solid var(--fs-search-dropdown-border-color);
+      border-top: none;
+      border-radius: var(--fs-search-dropdown-border-radius);
+      box-shadow: var(--fs-search-dropdown-box-shadow);
+    }
+
+    [data-fs-search-dropdown-loading-text] { padding: var(--fs-spacing-3); }
+
+    section, [data-fs-search-section] {
+      border-bottom: var(--fs-border-width) solid var(--fs-border-color-light);
```
Comment (renatamottam) [inline packages/ui/src/components/molecules/SearchDropdown/styles.scss:58]: I think we should create a `--fs-search-dropdown-section-border-color` for the `var(--fs-border-color-light)`

```suggestion
      border-bottom: var(--fs-search-dropdown-border-width) solid var(--fs-search-dropdown-section-border-color);
```

**Review by renatamottam — COMMENTED** [review]
Amazing work, fan! Left a few comments below

File: packages/components/src/molecules/SearchInputField/SearchInputField.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    >
+      <Input
+        ref={inputRef}
+        aria-label={ariaLabel}
+        data-fs-search-input-field-input
+        {...otherProps}
+      />
+      <Button type="submit" aria-label="Submit Search" {...buttonProps}>
+        <Icon component={buttonIcon ?? <MagnifyingGlass />} />
+      </Button>
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply packages/components/src/molecules/SearchInputField/SearchInputField.tsx:92]: great!

File: packages/core/src/components/search/SearchDropdown/SearchDropdown.tsx [hunk 4]
```diff
@@ -1,43 +1,54 @@
...
+            suggestion={suggestion}
+            linkProps={{
+              href: formatSearchPath(suggestion),
+              onClick: () =>
+                onSearchSelection?.(suggestion, formatSearchPath(suggestion)),
+            }}
+          />
+        ))}
+      </UISearchAutoComplete>
+      <SearchProducts data-fs-search-section>
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply packages/core/src/components/search/SearchDropdown/SearchDropdown.tsx:39]: nops! I will remove it!

File: apps/site/pages/components/molecules/search-dropdown.mdx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+  return {
+    SearchDropdown: <PropsSection propsList={searchDropdownProps} />,
+  }[component]
+}
+
+<header>
+
+# Search Dropdown
+
+Combine `SearchHistory`, `SearchTop` and `SearchAutoComplete` and `SearchProducts` to show all options in `SearchInput`. This component is part of the [Search](/components/search) feature.
```
Comment (eduardoformiga) [inline apps/site/pages/components/molecules/search-dropdown.mdx:49]:
```suggestion
Combine `SearchHistory`, `SearchTop`, `SearchAutoComplete` and `SearchProducts` to show all options in `SearchInput`. This component is part of the [Search](/components/search) feature.
```

File: apps/site/components/Search/SearchDropdownUsage.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  term: string
+  products?: [] | null
+}
+
+const SearchHistoryUsage = () => {
+  return (
+    <SearchHistory title="History">
+      <SearchHistoryTerm value="Headphone" linkProps={{ href: '/' }} />
+      <SearchHistoryTerm value="Audio & Video" linkProps={{ href: '/' }} />
+      <SearchHistoryTerm value="mh-7000" linkProps={{ href: '/' }} />
```
Comment (lucasfp13) [inline apps/site/components/Search/SearchDropdownUsage.tsx:29]: Maybe using `#` as href will avoid the page redirect if user tries to click on, but it's literally **not** a blocker

**Review by lucasfp13 — APPROVED** [review]
LGTM, tested through the docs.
Great job ✨

**Review by eduardoformiga — APPROVED** [review]
Awesome job! the Search docs are amazing ✨ 

In my tests everything works fine in the store and in the docs as well.
I'm not sure if the components package is the best place to include some hooks, as we talked. But looks good for now.

File: apps/site/components/Search/SearchDropdownUsage.tsx (hunk 6 above)
Comment (hellofanny) [reply apps/site/components/Search/SearchDropdownUsage.tsx:29]: agreed!!!

**Pull request merged**